Geometry/HcalCommonData/src/HcalDDDRecConstants.cc

```diff
@@ -861,7 +864,7 @@ void HcalDDDRecConstants::initialize(void) {
     int             phi    = (phis[0] > 0) ? phis[0] : -phis[0];
     int             zside  = (phis[0] > 0) ? 1 : -1;
     HcalSubdetector subdet = (subdet0 == static_cast<int>(HcalBarrel)) ? HcalBarrel : HcalEndcap;
```

Comment:
@bsunanda - this is an example where strongly typed enum would allow to drop subdet0 variable. The line 860:
```
int subdet0 = hcons.ldMap()->validDet(phis);
```
would become:
```
HcalSubdetector subdet = HcalSubdetector( hcons.ldMap()->validDet(phis));  // explicit conversion
```
the lines 862 and 863:
```
if ( subdet == HcalBarrel || subdet == HcalEndcap ) {
```
and the line 866 would be removed.